core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsTests.java

```diff
@@ -941,6 +956,47 @@ public void testNodeNotReachableFromMaster() throws Exception {
         ensureStableCluster(3);
     }
 
+    @Test
+    public void testIndexImportedFromDataOnlyNodesIfMasterLostDataFolder() throws Exception {
+        // test for https://github.com/elastic/elasticsearch/issues/8823
+        configureCluster(2, 1);
+        String masterNode = internalCluster().startMasterNode(Settings.EMPTY);
+        internalCluster().startDataOnlyNode(Settings.EMPTY);
+
+        ensureStableCluster(2);
+        assertAcked(prepareCreate("index").setSettings(Settings.builder().put("index.number_of_replicas", 0)));
+        index("index", "doc", "1", jsonBuilder().startObject().field("text", "some text").endObject());
+        ensureGreen();
+
+        internalCluster().restartRandomNode(new InternalTestCluster.MasterNodePredicate(masterNode), new InternalTestCluster.RestartCallback() {
+            public boolean clearData(String nodeName) {
+                return true;
+            }
+        });
+
+        ensureGreen("index");
+        assertTrue(client().prepareGet("index", "doc", "1").get().isExists());
+    }
+
+    // tests if indices are really deleted even if a master transition inbetween
+    @Ignore("https://github.com/elastic/elasticsearch/issues/11665")
+    @Test
+    public void testIndicesDeleted() throws Exception {
+        configureCluster(3, 2);
```

Comment:
I changed this back because of this comment: https://github.com/elastic/elasticsearch/pull/11666#discussion_r35305032